### PR TITLE
Bump elastic stack version automation

### PR DESCRIPTION
**Pull request by v1v**
## What does this PR do?

This PR is one part of the required changes to enable the automation. The bump automation for this project is only in charge to:

1) Create the PR with the new pinned version.
  a) Files changes are only affecting the test environments
  b) and the version validation.

Then the team is responsible to validate if the changes are good to be merged, for instance, whether it's needed a new change in the kibana-config.yml file or anything similar. 

## Why is it important?

Handle the stack versions on a PR basis to ensure breaking changes are handled correctly. Unfortunately we cannot use `dependabot` since those versions are handled differently.

## Issues

Similar to https://github.com/elastic/package-registry/pull/691

## Questions

- What's the backport policy for the `snapshot`, `staging` and `production` branches? Can I backport this PR if it gets merged?

## Follow ups

- Enable automerge with `mergify`?
- This project uses a different branch strategy so there is a requirement to match the branches with the elastic stack branch strategy (7.x, 7.<minor>, master=8.0.0...)

## Issues

https://github.com/elastic/package-storage/pull/1451 was created with the script from this PR

## Tests

Script to bump the versions:

```bash
$ .ci/bump-stack-version.sh 8.0.0-3beeb19e false
Branch creation disabled.
Update stack with version 8.0.0-3beeb19e
Commit changes
[feature/bump-stack-version-automation-snapshot 7cba2c48] bump stack version 8.0.0-3beeb19e
 2 files changed, 4 insertions(+), 4 deletions(-)
commit 7cba2c480a0934e645e16278a58fa706d820e5db (HEAD -> feature/bump-stack-version-automation-snapshot)
Author: Victor Martinez <VictorMartinezRubio@gmail.com>
Date:   Fri Jun 18 10:51:06 2021 +0100

    bump stack version 8.0.0-3beeb19e
You can now push and create a Pull Request
```

produces:

```diff
diff --git a/testing/environments/snapshot.yml b/testing/environments/snapshot.yml
index 754b97bc..4da3189d 100644
--- a/testing/environments/snapshot.yml
+++ b/testing/environments/snapshot.yml
@@ -3,7 +3,7 @@
 version: '2.3'
 services:
   elasticsearch:
-    image: docker.elastic.co/elasticsearch/elasticsearch:7.13.0-SNAPSHOT
+    image: docker.elastic.co/elasticsearch/elasticsearch:8.0.0-3beeb19e-SNAPSHOT
     healthcheck:
       test: ["CMD", "curl", "-f", "-u", "elastic:changeme", "http://127.0.0.1:9200/"]
       retries: 300
@@ -24,7 +24,7 @@ services:
     - "script.context.template.cache_max_size=2000"
 
   kibana:
-    image: docker.elastic.co/kibana/kibana:7.13.0-SNAPSHOT
+    image: docker.elastic.co/kibana/kibana:8.0.0-3beeb19e-SNAPSHOT
     depends_on:
       elasticsearch:
         condition: service_healthy
@@ -38,7 +38,7 @@ services:
       - ./kibana.config.yml:/usr/share/kibana/config/kibana.yml
 
   fleet-server:
-    image: docker.elastic.co/beats/elastic-agent:7.13.0-SNAPSHOT
+    image: docker.elastic.co/beats/elastic-agent:8.0.0-3beeb19e-SNAPSHOT
     depends_on:
       elasticsearch:
         condition: service_healthy
diff --git a/testing/main_integration_test.go b/testing/main_integration_test.go
index a4ba7907..6c84421c 100644
--- a/testing/main_integration_test.go
+++ b/testing/main_integration_test.go
@@ -112,7 +112,7 @@ type Package struct {
 
 func getPackages(t *testing.T) ([]string, error) {
 	// The kibana.version must be in sync with the stack version used in snapshot.yml
-	resp, err := http.Get("http://localhost:8080/search?experimental=true&kibana.version=7.13.0")
+	resp, err := http.Get("http://localhost:8080/search?experimental=true&kibana.version=8.0.0")
 	require.NoError(t, err)
 	defer resp.Body.Close()
```

### CODE REVIEW

File: testing/environments/snapshot.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@
 version: '2.3'
 services:
   elasticsearch:
-    image: docker.elastic.co/elasticsearch/elasticsearch:7.11.0-SNAPSHOT
+    image: docker.elastic.co/elasticsearch/elasticsearch:8.0.0-SNAPSHOT
```
Comment (mtojek) [inline testing/environments/snapshot.yml:6]: I'm afraid it won't work this way. First you need to upgrade the stack to support the Fleet Server. For reference: https://github.com/elastic/elastic-package/blob/master/internal/install/static_snapshot_yml.go

Comment (mtojek) [reply testing/environments/snapshot.yml:6]: It also seems to be a similar PR to @ycombinator 's one: https://github.com/elastic/package-storage/pull/1205

**Comment by ycombinator** [comment]
@v1v I recently merged #1205 so you'll need to rebase this PR on the `snapshot` branch to resolve conflicts.

File: testing/environments/snapshot.yml (hunk 1 above)
Comment (v1v) [reply testing/environments/snapshot.yml]: I'll syncup with you sometime this week then we can see what's required to apply a bump automation in this project and the others

**Review by cachedout — APPROVED** [review]

**Comment by v1v** [comment]
https://github.com/elastic/package-storage/issues/364 will remove all the ITs eventually, so this approach is not needed

**Pull request closed without merging**